pkg/reloader/vault.go

```diff
@@ -0,0 +1,193 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reloader
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/bank-vaults/vault-sdk/vault"
+	vaultapi "github.com/hashicorp/vault/api"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logrusadapter "logur.dev/adapter/logrus"
+)
+
+type VaultConfig struct {
+	Addr                 string
+	AuthMethod           string
+	Role                 string
+	Path                 string
+	Namespace            string
+	SkipVerify           bool
+	TLSSecret            string
+	TLSSecretNS          string
+	ClientTimeout        time.Duration
+	IgnoreMissingSecrets bool
+}
+
+func getVaultConfigFromEnv() *VaultConfig {
+	var vaultConfig VaultConfig
+
+	vaultConfig.Addr = os.Getenv("VAULT_ADDR")
+	if vaultConfig.Addr == "" {
+		vaultConfig.Addr = "https://vault:8200"
+	}
+
+	vaultConfig.AuthMethod = os.Getenv("VAULT_AUTH_METHOD")
+	if vaultConfig.AuthMethod == "" {
+		vaultConfig.AuthMethod = "jwt"
+	}
+
+	vaultConfig.Role = os.Getenv("VAULT_ROLE")
+
+	vaultConfig.Path = os.Getenv("VAULT_PATH")
+	if vaultConfig.Path == "" {
+		vaultConfig.Path = "kubernetes"
+	}
+
+	vaultConfig.Namespace = os.Getenv("VAULT_NAMESPACE")
+	if vaultConfig.Namespace == "" {
+		vaultConfig.Namespace = "default"
+	}
+
+	vaultConfig.SkipVerify, _ = strconv.ParseBool(os.Getenv("VAULT_SKIP_VERIFY"))
+
+	vaultConfig.TLSSecret = os.Getenv("VAULT_TLS_SECRET")
+
+	vaultConfig.TLSSecretNS = os.Getenv("VAULT_TLS_SECRET_NS")
+	if vaultConfig.TLSSecretNS == "" {
+		vaultConfig.TLSSecretNS = "default"
+	}
+
+	vaultConfig.ClientTimeout, _ = time.ParseDuration(os.Getenv("VAULT_CLIENT_TIMEOUT"))
+	if vaultConfig.ClientTimeout == 0 {
+		vaultConfig.ClientTimeout = 10 * time.Second
+	}
+
+	vaultConfig.IgnoreMissingSecrets, _ = strconv.ParseBool(os.Getenv("VAULT_IGNORE_MISSING_SECRETS"))
+
+	return &vaultConfig
+}
+
+func (c *Controller) initVaultClient() error {
+	actionWordStub := "initializ"
+
+	if c.vaultClient != nil {
+		_, err := c.vaultClient.Sys().Health()
+		if err == nil {
+			// Client is valid, no need to init
+			return nil
+		}
+		// log error and continue with (re)creating client
+		c.logger.Error("connection to Vault lost")
+		actionWordStub = "reinitializ"
+	}
+
+	errorMsg := fmt.Sprintf("failed to %s Vault client: ", actionWordStub+"e")
+	caser := cases.Title(language.AmericanEnglish)
+	c.logger.Infof("%s Vault client", caser.String(actionWordStub+"ing"))
```

Comment:
```suggestion
	if c.vaultClient != nil {
		_, err := c.vaultClient.Sys().Health()
		if err == nil {
			// Client is valid, no need to init
			return nil
		}
		// log error and continue with (re)creating client
		c.logger.Error("connection to Vault lost, recreating client")
	} else {
		c.logger.Infof("initializing Vault client")
	}
```

Reply:
No necessary for just a simple log message. Don't forget to remove the imports as well.